### DISCUSSION TITLE
Guidance on ignoring dependencies with prefixes

### DISCUSSION THREAD

**Discussion by tomkerkhove** [renovatebot/renovate#16731] [Request Help]
We want to ignore certain dependencies with a prefix (`mcr.microsoft.com/mirror/`) but it seems like Renovate still tries to pull metadata for them which are causing failures.

Our current config has the following package rules:
```json
"packageRules": [
    {
      "datasources": ["docker"],
      "excludePackagePrefixes": ["mcr.microsoft.com/mirror/"],
      "groupName": "Docker images"
    },
    {
      "datasources": ["docker"],
      "matchPackagePrefixes": ["mcr.microsoft.com/dotnet/"],
      "groupName": ".NET Core Docker containers"
    }
  ],
```

However, it still seems to try to pull the metadata:

> WARN: Invalid registry response (repository=name)
>        "apiCheckUrl": "[https://mcr.microsoft.com/v2/mirror/docker/library/python/tags/list?n=10000",](https://mcr.microsoft.com/v2/mirror/docker/library/python/tags/list?n=10000%22,)
>        "res": {
>          "statusCode": 404,

Are the package rules wrong or does Renovate always pull metadata? If so, can we exclude container images?

**Comment by viceice** [comment]
you need to add a new package rule to disable those deps. by default renovate pulls updates for each dependency

**Reply by tomkerkhove** [reply]
Thanks, that's what I was looking for!

Turns out that enabled is what I need, thanks a ton! 

```json
{
  "packageRules": [
    {
      "matchPackageNames": ["eslint", "eslint-config-base"],
      "enabled": false
    }
  ]
}
```

**Comment by rarkins** [comment]
Why would you expect your current config to be ignoring anything at all? You just set a group name, so I'm not sure why you're surprised Renovate attempted to look them up unless you left out some config above

**Reply by tomkerkhove** [reply]
I figure if it will be marked as excluded that it won't look it up but the default behavior is to update all found dependencies which I forgot.

That explains the behavior though. Given ignoreDeps doesn't support patterns I'll need to go with something similar to the documented workaround :

```json
{
  "packageRules": [
    {
      "matchPackageNames": ["eslint", "eslint-config-base"],
      "enabled": false
    }
  ]
}
```
